datastream-server/src/main/java/com/linkedin/datastream/server/Coordinator.java

```diff
@@ -716,11 +716,11 @@ private void handleAssignmentChange(boolean isDatastreamUpdate) throws TimeoutEx
 
     if (_config.getEnableAssignmentTokens()) {
       try {
+        ExecutorService executor = Executors.newSingleThreadExecutor();
```

Comment:
I know that we don't follow this practice from the beginning, but we started this once we got feedback to have meaningful name in the threaddump and looking at review process of Samza/Cruise Control.

Please use the other API which helps with naming the executor thread. 

For eg:

 ` // An executor to spawn threads to close the producer.
  private final ExecutorService _producerCloseExecutorService = Executors.newSingleThreadExecutor(
      new ThreadFactoryBuilder().setNameFormat("KafkaProducerWrapperClose-%d").build());`